sklearn/preprocessing/_data.py

```diff
@@ -1050,7 +1056,7 @@ def partial_fit(self, X, y=None, sample_weight=None):
         # for backward-compatibility, reduce n_samples_seen_ to an integer
         # if the number of samples is the same for each feature (i.e. no
         # missing values)
-        if np.ptp(self.n_samples_seen_) == 0:
+        if xp.max(self.n_samples_seen_) == xp.min(self.n_samples_seen_):
```

Comment:
FYI, I double-checked and `np.ptp` does nothing smart (like computing both the `min` and `max` in a single pass) so this is fine to replace it like this.